ert_data/measured.py

```diff
@@ -0,0 +1,129 @@
+
+    @property
+    def data(self):
+        return self._data
+
+    def _set_data(self, data):
+        expected_keys = ["OBS", "STD"]
+        if not isinstance(data, pd.DataFrame):
+            raise TypeError(
+                "Invalid type: {}, should be type: {}".format(type(data), pd.DataFrame)
+            )
+        elif not set(expected_keys).issubset(data.index):
+            raise ValueError(
+                "{} should be present in DataFrame index, missing: {}".format(
+                    ["OBS", "STD"], set(expected_keys) - set(data.index)
+                )
+            )
+        else:
+            self._data = data
+
+    def remove_failed_realizations(self):
+        self._set_data(self._remove_failed_realizations())
+
+    def _remove_failed_realizations(self):
+        """Removes rows with no simulated data, leaving observations and
+        standard deviations as-is."""
+        pre_index = self.data.index
+        post_index = list(self.data.dropna(axis=0, how="all").index)
+        drop_index = set(pre_index) - set(post_index + ["STD", "OBS"])
+        return self.data.drop(index=drop_index)
+
+    def remove_inactive_observations(self):
+        self._set_data(self._remove_inactive_observations())
+
+    def _remove_inactive_observations(self):
+        """Removes columns with one or more NaN values."""
+        filtered_dataset = self.data.dropna(axis=1)
+        if filtered_dataset.empty:
+            raise ValueError(
+                "This operation results in an empty dataset (could be due to one or more failed realizations)"
+            )
+        return filtered_dataset
+
+    def is_empty(self):
+        return self.data.empty
+
+    def _get_data(self, observation_keys, index_lists):
+        """
+        Adds simulated and observed data and returns a dataframe where ensamble
+        members will have a data key, observed data will be named OBS and
+        observed standard deviation will be named STD.
+        """
+        measured_data = pd.DataFrame()
+        case_name = self._facade.get_current_case_name()
+
+        if index_lists is None:
+            index_lists = [None] * len(observation_keys)
+
+        if len(index_lists) != len(observation_keys):
+            raise ValueError("index list must be same length as observations keys")
+
+        for key, index_list in zip(observation_keys, index_lists):
+            observation_type = self._facade.get_impl_type_name_for_obs_key(key)
+            data_loader = loader.data_loader_factory(observation_type)
+
+            data = data_loader(self._facade, key, case_name)
+            data = MeasuredData.filter_on_column_index(data, index_list)
+            data = pd.concat({key: data}, axis=1)
+
+            measured_data = pd.concat([measured_data, data], axis=1)
+
+        return measured_data
+
+    def filter_ensamble_std(self, std_cutoff):
+        self._set_data(self._filter_ensamble_std(std_cutoff))
+
+    def filter_ens_mean_obs(self, alpha):
```

Comment:
I would prefer if we spelled out ensemble, or at the very least that we are consistent in our API regarding this ;)

Reply:
You're right, was a bit confusing, fixed